src/OVAL/probes/unix/linux/dpkginfo_probe.c

```diff
@@ -92,8 +93,9 @@ void dpkginfo_probe_fini (void *ptr)
 {
         struct dpkginfo_global *d = (struct dpkginfo_global *)ptr;
 
+        pthread_mutex_lock (&(d->mutex));
         dpkginfo_fini();
-        pthread_mutex_destroy (&(d->mutex));
+        pthread_mutex_unlock (&(d->mutex));
```

Comment:
Why don't we reset the `.init_done` here? The global struct is now uninitialized, right?

Reply:
The `dpkginfo_global` structure is initialized statically. `.init_done` is set to `1` once `dpkginfo_init` has been called for the first time. `dpkginfo_fini` only calls `cgCache->Close`, so `cgCache->ReadOnlyOpen` could update the dpkg cache the next time `dpkginfo_get_by_name` will be called. `cgCache` is allocated once, when `dpkginfo_init`  has been called for the first time. Then, `cgCache` is never freed.